File: procuro_system/lib/models/Delivery.dart
class Delivery {
  static const ID = "_id";
  static const DELIVERYID = "deliveryId";
  static const DATE = "date";
  static const ORDERID = "orderId";
  static const SUPPLIERID = "supplierId";
  static const ITEMS = "items";
  static const COMMENT = "comment";

  final String id;
  final int deliveryId;
  final String date;
  final Object order;
  final Object supplier;
  final List<dynamic> items;
  final String comment;

  Delivery({
      required this.id,
      required this.deliveryId,
      required this.date,
      required this.order,
      required this.supplier,
      required this.items,
      required this.comment
  });

  factory Delivery.fromJson(Map<String, dynamic> json) {
    var delivery = Delivery(
      id: json[Delivery.ID],
      deliveryId: json[Delivery.DELIVERYID],
      date: json[Delivery.DATE],
      order: json[Delivery.ORDERID],
      supplier: json[Delivery.SUPPLIERID],
      items: json[Delivery.ITEMS],
      comment: json[Delivery.COMMENT],
    );
    return delivery;
  }
}

class DeliveryList {
  final List<Delivery> deliveries;

  DeliveryList({required this.deliveries});

  factory DeliveryList.fromJson(List<dynamic> parsedJson) {
    List<Delivery> deliveries = <Delivery>[];
    deliveries = parsedJson.map((i) => Delivery.fromJson(i)).toList();
    return new DeliveryList(deliveries: deliveries);
  }
}

File: Backend/src/models/Delivery.ts
import { IDelivery } from '../interfaces/IDelivery';
import * as mongoose from 'mongoose';

const { Schema } = mongoose;

const DeliverySchema = new Schema({
    deliveryId: {
        type: Number,
        required: true
    },
    date: {
        type: Date,
        required: true
    },
    orderId: {
        type: Schema.Types.ObjectId,
        ref: 'orders',
        required: true
    },
    supplierId: {
        type: Schema.Types.ObjectId,
        ref: 'supplier',
        required: true
    },
    items: {
        type: Array,
        required: true,
    },
    comment: {
        type: String,
        trim: true,
        required: false
    }
}, { timestamps: true });

export default mongoose.model<IDelivery & mongoose.Document>('deliveries', DeliverySchema);
